src/app/pages/common/entity/utils.spec.ts

```diff
@@ -0,0 +1,866 @@
+    ],
+  },
+  {
+    name: 'should parse "show_if"',
+    schema: {
+      variable: 'updateStrategy',
+      description: 'Upgrade Policy',
+      label: 'Update Strategy',
+      group: 'Scaling/Upgrade Policy',
+      schema: {
+        type: 'string',
+        show_if: [
+          ['workloadType', '=', 'Deployment'],
+        ],
+        default: 'RollingUpdate',
+        enum: [
+          {
+            value: 'RollingUpdate',
+            description: 'Create new pods and then kill old ones',
+          },
+          {
+            value: 'Recreate',
+            description: 'Kill existing pods before creating new ones',
+          },
+        ],
+      },
+    },
+    expectedFieldConfigs: [
+      {
+        name: 'updateStrategy',
+        value: 'RollingUpdate',
+        tooltip: 'Upgrade Policy',
+        placeholder: 'Update Strategy',
+        type: 'select',
+        options: [
+          {
+            value: 'RollingUpdate',
+            label: 'Create new pods and then kill old ones',
+          },
+          {
+            value: 'Recreate',
+            label: 'Kill existing pods before creating new ones',
+          },
+        ],
+        relation: [
+          {
+            action: 'SHOW',
+            when: [{ name: 'workloadType', operator: '=', value: 'Deployment' }],
+          },
+        ],
+      },
+    ],
+  },
+];
+
+const unsupportedTypeData = {
+  variable: 'cronSchedule',
+  label: 'Cron Schedule',
+  group: 'Workload Details',
+  schema: {
+    type: 'cron',
+    default: {
+      minute: '5',
+    },
+  },
+};
+
+describe('parseSchemaFieldConfig', () => {
+  const utils = new EntityUtils();
+
+  describe.each(basicTypeData)('parse the basic types', (item) => {
+    it(item.name, () => {
+      const result = utils.parseSchemaFieldConfig(item.schema);
+      const isEqual = _.isEqual(item.expectedFieldConfigs, result);
+      expect(isEqual).toBe(true);
+    });
+  });
+
+  describe.each(nestedData)('parse the nested data', (item) => {
+    it(item.name, () => {
+      const result = utils.parseSchemaFieldConfig(item.schema);
+      const isEqual = _.isEqual(item.expectedFieldConfigs, result);
+      expect(isEqual).toBe(true);
+    });
+  });
+
+  describe.each(propertyData)('parse the property(hidden, readonly)', (item) => {
+    it(item.name, () => {
+      const result = utils.parseSchemaFieldConfig(item.schema);
+      const isEqual = _.isEqual(item.expectedFieldConfigs, result);
+      expect(isEqual).toBe(true);
+    });
+  });
+
+  describe.each(relationData)('parse the relations', (item) => {
+    it(item.name, () => {
+      const result = utils.parseSchemaFieldConfig(item.schema);
+      const isEqual = _.isEqual(item.expectedFieldConfigs, result);
+      expect(isEqual).toBe(true);
+    });
+  });
+
+  describe('parse unexpected type', () => {
+    it('should return empty and show error on console', () => {
+      const originalError = console.error;
+      console.error = jest.fn();
```

Comment:
A more stream lined way to deal with errors might be to use expect(fn).toThrow()

https://jestjs.io/docs/expect#tothrowerror

Reply:
@dszidi this is for testing console.error, `console.error('Unsupported type=', schemaConfig);`.
I think `expect(fn).toThrow()` can not test this case properly, because console.error does not throw an error.

Reply:
Ok, sounds good